Math/NumberTheory/Diophantine.hs

```diff
@@ -64,3 +74,35 @@ cornacchia d m
  where
   candidates = map (\sf -> (sf, m `div` (sf * sf))) (squareFactors m)
   solve (sf, m') = map (\(x, y) -> (x * sf, y * sf)) (cornacchiaPrimitive d m')
+
+----
+
+-- | A linear diophantine equation `ax + by = c`
+-- | where `x` and `y` are unknown
+data Linear a = Lin { a,b,c :: a }
```

Comment:
This makes `a`, `b` and `c` globally defined functions, stealing them from any other potential usage. Could you please use something more specific and less likely to clash? Same applies to `LinearSolution`.

Reply:
If that is a problem I am more inclined to not have names at all, or something like `a_var`? The current names are what is used in the literature

Reply:
I am not as attached to the names in `LinearSolution`. The only reason I have it rather than returning `Maybe (Int -> (Int, Int))` is that `runLinearSolution` is a very simple function, and applying it is technically lossy

Reply:
Sorry for the spam.

I am leaning towards just removing `Linear`. It is only used in one function- the user can create their own data type if they really want it.

On the other hand, I know that `Linear` has a sensible `Semigroup` instance, and many analogues of `liftA2 f` makes sense for it:
```haskell
-- Basically `liftA2 (+)`
-- (a1+a2)*x + (b1+b2)*y = c1 + c2
--
add (Lin a1 b1 c1) (Lin a2 b2 c2) =
  Lin (a1+a2) (b1+b2) (c1+c2)
```
I just didn't add instances because I didn't want to bloat this PR.